dissect/target/plugins/os/unix/linux/proc.py

```diff
@@ -53,71 +53,68 @@ class NetSocket:
     tx_queue: int  # parsed value from tx_rx_queue
     pid: int  # pid of the socket
     name: str  # process name associated to the socket
-    cmdline: str  # process cmdline associated to the socket
-
-    def __init__(self):
-        for fld in fields(self):
-            setattr(self, fld.name, None)
+    cmdline: str = field(default=None)  # process cmdline associated to the socket
 
-    def from_line(self, line: str) -> NetSocket:
-        splits = line.split()
+    @classmethod
+    def from_line(cls, line: str) -> NetSocket:
+        socket = cls()
 
         # The lines in /proc/net/[protocol] can be of arbitrary column (split) length, but are ordered.
-        # So we use the fields(NetSocket) to dynamically fill the NetSockt dataclass.
-        for idx, value in enumerate(splits):
-            field_names = fields(self)
-            setattr(self, field_names[idx].name, value)
-
-        self.uid = int(self.uid)
-        self.inode = int(self.inode)
-        self.tx_queue, self.rx_queue = [int(queue, 16) for queue in self.tx_rx_queue.split(":")]
-        self.local_port, self.remote_port = [
+        # So we use the fields(NetSocket) to dynamically fill the NetSocket dataclass.
+        for idx, value in enumerate(line.split()):
+            field_names = fields(cls)
+            setattr(socket, field_names[idx].name, value)
+
+        socket.uid = int(socket.uid)
+        socket.inode = int(socket.inode)
+        socket.tx_queue, socket.rx_queue = [int(queue, 16) for queue in socket.tx_rx_queue.split(":")]
+        socket.local_port, socket.remote_port = [
             int(port, 16)
             for port in (
-                self.local_address.split(":")[1],
-                self.rem_address.split(":")[1],
+                socket.local_address.split(":")[1],
+                socket.rem_address.split(":")[1],
```

Comment:
I see there's a `local_ip` and `remote_ip` field in the dataclass but you don't fill them?